bin/ruby

```diff
@@ -0,0 +1,49 @@
+#!/bin/sh
+# Run a ruby script using the logstash jruby launcher
+#
+# Usage:
+#   bin/ruby [arguments]
+#
+# Supported environment variables:
+#   LS_JVM_OPTS="xxx" path to file with JVM options
+#   LS_JAVA_OPTS="xxx" to append extra options to the defaults JAVA_OPTS provided by logstash
+#   JAVA_OPTS="xxx" to *completely override* the defauls set of JAVA_OPTS provided by logstash
+#
+# Development environment variables:
+#   USE_RUBY=1 to force use the local "ruby" command to launch logstash instead of using the vendored JRuby
+#   DEBUG=1 to output debugging information
+
+# use faster starting JRuby options see https://github.com/jruby/jruby/wiki/Improving-startup-time
+export JRUBY_OPTS="$JRUBY_OPTS -J-XX:+TieredCompilation -J-XX:TieredStopAtLevel=1"
+
+# sourcepath discovery below is the same as in bin/logstash
+
+unset CDPATH
+# This unwieldy bit of scripting is to try to catch instances where Logstash
+# was launched from a symlink, rather than a full path to the Logstash binary
+if [ -L $0 ]; then
```

Comment:
we check if $0 here is a symlink, but if the path contains a space, it fails:

```
% "/tmp/hello world/bin/ruby" --version
/tmp/hello world/bin/ruby: line 24: [: /tmp/hello: binary operator expected
```

This is what I refer to with respect to quoting. If you don't quote your variables, any value with a space will be buggy.